spec/models/solr_document_tree_spec.rb

```diff
@@ -0,0 +1,53 @@
+# frozen_string_literal: true
+
+require "rails_helper"
+
+RSpec.describe SolrDocumentTree do
+  subject(:solr_document_tree) { described_class.new(root: solr_document) }
+
+  let(:indexer_settings) do
+    {
+      repository: "publicpolicy"
+    }
+  end
+  let(:indexer) do
+    Traject::Indexer::NokogiriIndexer.new(indexer_settings).tap do |i|
+      i.load_config_file(Rails.root.join("lib", "pulfalight", "traject", "ead2_config.rb"))
+    end
+  end
+  let(:fixture_file) do
+    File.read(fixture_path)
+  end
+  let(:nokogiri_reader) do
+    Arclight::Traject::NokogiriNamespacelessReader.new(fixture_file.to_s, indexer.settings)
+  end
+  let(:records) do
+    nokogiri_reader.to_a
+  end
+  let(:record) do
+    records.first
+  end
+  let(:values) do
+    indexer.map_record(record)
+  end
+  let(:solr_document) do
+    SolrDocument.new(values)
+  end
+
+  context "when the Solr Document contains one or more member components" do
+    let(:fixture_path) do
+      Rails.root.join("spec", "fixtures", "aspace", "generated", "publicpolicy", "MC152.EAD.xml")
```

Comment:
No need to index this record again. It's already a fixture. On line 34, you could just use `SolrDocument.find("MC152")`